improve SEO on statements

BillWagner opened the pull request:
In all the statements, operators, and tokens pages, edit the title, description, and opening paragraph to increase SEO on the statement keywords.

First commit - edit all statements pages.
Second commit - edit all the operator pages.

Fixes #32497

For this issue, update the title, description, and opening paragraph to include the keyword, name of operators, the symbols, and the term used for each operator. Where the opening was a list of the operators, convert that to a paragraph so it shows in search results. Also, do a general edit pass.

For review, checking the opening two paragraphs should be sufficient. 

Also check modified pages for open issues. Fix the following:

#32630 - see Language reference [lambda expressions attributes](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/lambda-expressions?branch=pr-en-us-32704#attributes).
#26112 - see Language reference operators See [operator associativity](https://review.learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/?branch=pr-en-us-32704#operator-associativity).

BillWagner committed:
improve SEO on statements

In all the statements pages, edit the title, description, and opening paragraph to increase SEO on the statement keywords.

BillWagner committed:
Update all the operator pages

BillWagner committed:
Improve SEO for tokens

The special characters `@` and `$` are hard to search for. Include the names in the title and descriptions to improve possible search terms.

BillWagner committed:
fix warnings

pkulikov commented on docs/csharp/language-reference/operators/patterns.md:
```diff
@@ -12,28 +12,9 @@ helpviewer_keywords:
-- [List patterns](#list-patterns): to test if sequence elements match corresponding nested patterns. Introduced in C# 11.
-
-[Logical](#logical-patterns), [property](#property-pattern), [positional](#positional-pattern), and [list](#list-patterns) patterns are *recursive* patterns. That is, they can contain *nested* patterns.
+The following C# expressions and statements support pattern matching. The [`is` expression](is.md), the [switch statement](../statements/selection-statements.md#the-switch-statement) and the [switch expression](switch-expression.md). In those constructs, you can match an input expression against many patterns. The [declaration pattern](#declaration-and-type-patterns) checks the run-time type of a ...
```
Combined all in one paragraph and without listing, this te

BillWagner commented:
@pkulikov 

> could you please elaborate on the necessity of this

The most impactful motivation for this set of changes is [this query](https://www.google.com/search?q=C%23+if+statement).  

> To me, dropping list formatting reduces the readability. I wonder what has made such a trade-off worth it.

I'll take another look on the review site. The first lists I saw were small, and it worked well. I'll give another look at the longer ones. I think for the longer lists, I'll add the keywords in the first paragraph and keep the list for readability. That might be the best presentation.

pkulikov commented:
@BillWagner I see, thank you for the query example.

BillWagner committed:
restore some lists for readability

In some of the cases, the lists of statements or operators is a much more readable format than the prose. In those cases, expand the opening paragraph to include the keywords for likely search terms, and restore the list.

BillWagner commented:
> I see, thank you for the query example.

@pkulikov And thanks for pointing out the readability issue. I think this next version is much easier to read.

pkulikov commented:
@BillWagner yes, the updated version reads easier, thank you!

IEvangelist reviewed (approved):
This is great, well done! I should work SEO updates for some of the fundamentals too.
